src/compas_fab/robots/robot.py

```diff
@@ -1333,6 +1453,19 @@ def draw(self):
         """
         return self.draw_visual()
 
+    def draw_end_effector(self, tool0_frame):
```

Comment:
I believe this method should only be dispatching to the artist (also, why does it need `tool0_frame` to be specified?), e.g.:

```python
def draw_attached_tool(self):
    self.artist.draw_attached_tool()
```

and then on the `BaseRobotArtist`, the mesh transformation can be done with native transformations with something like:
```python
def draw_attached_tool(self):
    if not self.robot.attached_tool:
        return

    native_geometry = self.draw_geometry(sefl.robot.attached_tool.mesh)
    # here get the correct frame for the transformation
    # I would expect this to be `self.robot.get_end_effector_frame`, but I'm not sure why instead you had `tool0_frame` here.
    T = Transformation.from_frame_to_frame(self.robot.attached_tool.frame, self.robot.get_end_effector_frame())
    self.transform(native_geometry, T)
    return native_geometry
```

Reply:
Q: Should the artist or the robot also handle scaling of the attached tool?